cli/src/main/java/org/eclipse/hono/cli/app/ReceiverStatistics.java

```diff
@@ -0,0 +1,268 @@
+        this.receiver = Objects.requireNonNull(receiver);
+        this.vertx = Objects.requireNonNull(vertx);
+    }
+
+    /**
+     * Starts this component.
+     * <p>
+     * 
+     * @return A future indicating the outcome of the startup process.
+     */
+    @PostConstruct
+    Future<CompositeFuture> start() {
+        vertx.setPeriodic(interval, this::statistic);
+        receiver.setMessageHandler((endpoint, msg) -> handleMessage(endpoint, msg));
+        return Future.succeededFuture();
+    }
+
+    private void handleMessage(final String endpoint, final Message msg) {
+        final String deviceId = MessageHelper.getDeviceId(msg);
+
+        final Buffer payload = MessageHelper.getPayload(msg);
+
+        if (LOG_STATISTIC.isInfoEnabled()) {
+            final long now = System.nanoTime();
+            final Statistic total;
+            Statistic current;
+            synchronized (this) {
+                if (this.total == null) {
+                    this.total = new Statistic(now);
+                }
+                total = this.total;
+                if (this.current == null) {
+                    this.current = new PeriodStatistic(now, interval);
+                }
+                current = this.current;
+            }
+            total.increment(now);
+            if (!current.increment(now)) {
+                if (current.isPrinting()) {
+                    LOG_STATISTIC.info("statistic: total {}, last {}", total, current);
+                }
+                synchronized (this) {
+                    if (this.current == current) {
+                        this.current = new PeriodStatistic(now, interval);
+                    }
+                    current = this.current;
+                }
+                current.increment(now);
+            }
+        }
+
+        LOG_STATISTIC.trace("received {} message [device: {}, content-type: {}]: {}", endpoint, deviceId, msg.getContentType(),
+                payload);
+
+        if (msg.getApplicationProperties() != null) {
+            LOG_STATISTIC.trace("... with application properties: {}", msg.getApplicationProperties().getValue());
+        }
+    }
+
+    private void statistic(final Long id) {
+        final Statistic total;
+        final Statistic current;
+        synchronized (this) {
+            total = this.total;
+            current = this.current;
+        }
+        if (total != null) {
+            final long now = System.nanoTime();
+            if (current == null || current.finished(now)) {
+                if (current == null) {
+                    LOG_STATISTIC.info("statistic: total {}", total);
+                } else {
+                    LOG_STATISTIC.info("statistic: total {}, last {}", total, current);
+                }
+                synchronized (this) {
+                    if (this.total == total && this.current == current) {
+                        if (current == null) {
+                            this.total = null;
```

Comment:
This means that the `total` statistic is reset after having received no messages in a period of 2x interval, right?
I think this might be confusing when the user just looks at the last log line after a long test run, not seeing that the total number might not be the actual grand total because it got reset a few times.

Reply:
That may be more a question of your usage. I start usually that client and then I focus on my own test via the protocol adapters. Sometimes I adapt my test and want to see the changes. Without the reset I sometimes (sometimes very frequently :-) ), I forgot the restart the statistic client and as result, I had to repeat the test again with restart statistic client. though the test are more a couple of minutes, I found the auto-reset the better idea.

So, should I add a config flag for autoReset in order to be able to select the behaviour?

Reply:
Yes, I think that would be good.

Reply:
I added the config and disabled it by default.